task.ps1

```
@@ -7,17 +7,57 @@ $vnetAddressPrefix = "10.0.0.0/16"
-$nsgRuleSSH = New-AzNetworkSecurityRuleConfig -Name SSH  -Protocol Tcp -Direction Inbound -Priority 1001 -SourceAddressPrefix * -SourcePortRange * -DestinationAddressPrefix * -DestinationPortRange 22 -Access Allow;
-$nsgRuleHTTP = New-AzNetworkSecurityRuleConfig -Name HTTP  -Protocol Tcp -Direction Inbound -Priority 1002 -SourceAddressPrefix * -SourcePortRange * -DestinationAddressPrefix * -DestinationPortRange 8080 -Access Allow;
+$nsgRuleSSH = New-AzNetworkSecurityRuleConfig -Name SSH  -Protocol Tcp -Direction Inbound -Priority 1001 -SourceAddressPrefix * -SourcePortRange * -DestinationAddressPrefix * -DestinationPortRange 22 -Access Allow
+$nsgRuleHTTP = New-AzNetworkSecurityRuleConfig -Name HTTP -Protocol Tcp -Direction Inbound -Priority 1002 -SourceAddressPrefix * -SourcePortRange * -DestinationAddressPrefix * -DestinationPortRange 8080 -Access Allow
 New-AzNetworkSecurityGroup -Name $networkSecurityGroupName -ResourceGroupName $resourceGroupName -Location $location -SecurityRules $nsgRuleSSH, $nsgRuleHTTP
 
-# ↓↓↓ Write your code here ↓↓↓
+# ↓↓↓ Network + IP setup ↓↓↓
+$subnet  = New-AzVirtualNetworkSubnetConfig -Name $subnetName -AddressPrefix $subnetAddressPrefix
+$Vnet = New-AzVirtualNetwork -Name $virtualNetworkName -ResourceGroupName $resourceGroupName -Location $location -AddressPrefix $vnetAddressPrefix -Subnet $subnet
+$publicIp = New-AzPublicIpAddress -Name $publicIpAddressName -ResourceGroupName $resourceGroupName -AllocationMethod Static -DomainNameLabel $dnsPrefix -Location $location
+
+# SSH key resource (conditionally with/without public key)
+if ($sshKeyPublicKey) {
+    New-AzSshKey -ResourceGroupName $resourceGroupName -Name $sshKeyName -Location $location -PublicKey $sshKeyPublicKey
+} else {
+    Write-Host "⚠️ Public SSH key not found at $sshKeyPath. Creating SSH key resource without uploaded key."
+    New-AzSshKey -ResourceGroupName $resourceGroupName -Name $sshKeyName -Location $location
+}
```

Functional issue: when no local public SSH key is found you create the ssh key resource without a PublicKey and continue. If the SSH key resource has no public key, the VM may not be configured for SSH key authentication and you won't be able to SSH into the VM. Requirement 2.6 / 3.4 expects the VM to use SSH auth via the created SSH key resource. Consider either failing early with a clear error if no public key is available, or generating a keypair programmatically (e.g., run ssh-keygen) and use the generated public key with New-AzSshKey so the VM is provisioned with usable SSH auth.